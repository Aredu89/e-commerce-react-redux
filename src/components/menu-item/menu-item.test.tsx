import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import MenuItem from './menu-item.component';

const mockedNavigator = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedNavigator
}));

describe('Manu-item component', () => {
  const props = {
    title: 'Title',
    imageUrl: 'image/url',
    linkUrl: 'link/url',
    size: 'large'
  };
  it('renders properly', () => {
    expect(shallow(
      <BrowserRouter>
        <MenuItem {...props} />
      </BrowserRouter>
    )).toMatchSnapshot();
  })
});
