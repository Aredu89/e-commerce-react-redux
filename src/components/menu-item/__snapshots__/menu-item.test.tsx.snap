// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manu-item component renders properly 1`] = `ShallowWrapper {}`;
